src/rotation_conversions.jl

```diff
@@ -3,7 +3,7 @@ import Base: convert
 
 
 # a sgn function
-@inline sgn{T}(x::T) = copysign(T(1), x)  #hmm, x = -0.0
+@inline sgn{T}(x::T) = copysign(one(T), x)  # hmm, x = -0.0
```

Comment:
Its for stuff like quaternion * sgn(quaternion.realpart) to use the same
part of the quaternion ambiguity, where sign(0) == 0  is bad news

On 14 June 2016 at 10:51, Chris Foster <notifications@github.com> wrote:

> In src/rotation_conversions.jl
> <https://github.com/FugroRoames/Rotations.jl/pull/7#discussion_r66894014>:
>
> > @@ -3,7 +3,7 @@ import Base: convert
> >
> >
> >  # a sgn function
> > -@inline sgn{T}(x::T) = copysign(T(1), x)  #hmm, x = -0.0
> > +@inline sgn{T}(x::T) = copysign(one(T), x)  # hmm, x = -0.0
>
> Base.sign()?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FugroRoames/Rotations.jl/pull/7/files/a82fd30bae4568dd0441db4bcb705d39e0308dff#r66894014>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AQAxNNxEjngVgIeQ3WsDMGYKWy4HL4VVks5qLfsDgaJpZM4I026F>
> .
>